server/server.go

```diff
@@ -2734,6 +2761,17 @@ func (s *Server) ProfilerAddr() *net.TCPAddr {
 }
 
 func (s *Server) readyForConnections(d time.Duration) error {
+	// Wait for startup. At this point AcceptLoop will only just be
+	// starting, so we'll still check for the presence of listeners
+	// after this.
+	select {
```

Comment:
Not sure if this is that important, but maybe we should "deduct" from `d` the time spent in this select. Again, not sure if this is worth it.

Reply:
I suppose moving the `end := time.Now().Add(d)` declaration to before the `select` should have this effect.

Reply:
Probably.

Reply:
Have done this.